srcalc/src/main/java/gov/va/med/srcalc/domain/Calculation.java

```diff
@@ -177,16 +178,29 @@ void setValues(final SortedSet<Value> values)
      * Runs the calculation for each outcome with the given Values.
      * @throws IllegalArgumentException if incomplete values are provided
      * @return the outcomes for convenience
+     * @throws MissingValueListException if there are any required variables without an assigned value
      */
-    public SortedMap<String, Double> calculate(final Collection<Value> values)
+    public SortedMap<String, Double> calculate(final Collection<Value> values) throws MissingValueListException
     {
         // Run the calculation first to make sure we don't get any exceptions.
         final TreeMap<String, Double> outcomes = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        final List<MissingValueException> missingValues = new ArrayList<MissingValueException>();
         for (final RiskModel model : getSpecialty().getRiskModels())
         {
-            outcomes.put(model.getDisplayName(), model.calculate(values));
+            try
+            {
+            	outcomes.put(model.getDisplayName(), model.calculate(values));
+            }
+            catch(MissingValueListException e)
+            {
+            	missingValues.addAll(e.getMissingValues());
```

Comment:
Shouldn't this be a set instead? What if multiple models add the same exceptions? I'm OK with a TODO comment for now since we don't have to support multiple models until ASRC-46.